mri/reconstructors/self_calibrating.py

```diff
@@ -0,0 +1,180 @@
+    prox_op: operator, (optional default None)
+        Defines the proximity operator for the regularization function.
+        For example, for L1 Norm, the proximity operator is Thresholding
+        If None, the proximity opertaor is defined as Soft thresholding
+        of wavelet coefficients with mu value as specified.
+    mu: float or np.ndarray, (optional, default 0)
+        If prox_op is None, the value of mu is used to form a proximity
+        operator that is soft thresholding of the wavelet coefficients.
+        If prox_op is specified, this is ignored.
+    lips_calc_max_iter: int, default 10
+        Defines the maximum number of iterations to calculate the lipchitz
+        constant
+    num_check_lips: int, default 10
+        Number of iterations to check if the lipchitz constant is correct
+    lipschitz_cst: int, default None
+        The user specified lipschitz constant. If this is not specified,
+        it is calculated using PowerMethod
+    kspace_portion: int or tuple (default is 0.1 in all dimension)
+        int or tuple indicating the k-space portion used to estimate the coil
+        sensitivity information.
+        if int, will be evaluated to (0.1,)*nb_dim of the image
+    smaps_extraction_mode: string 'FFT' | 'NFFT' | 'Stack' | 'gridding' default
+        Defines the mode in which we would want to interpolate to extract the
+        sensitivity information,
+        NOTE: FFT should be considered only if the input has
+        been sampled on the grid
+    smaps_gridding_method: string 'linear' (default) | 'cubic' | 'nearest'
+        For gridding mode, it defines the way interpolation must be done used
+        by the sensitivity extraction method.
+    n_jobs: int, default 1
+        The number of CPUs used to accelerate the reconstruction
+    verbose: int, default 0
+        Verbosity level.
+            1 => Print basic debug information
+            5 => Print all initialization information
+            20 => Calculate cost at the end of each iteration.
+                NOTE : This is computationally intensive.
+            30 => Print the debug information of operators if defined by class
+    Note:
+    -----
+    The user is expected to specify the either prox_op or mu to obtain
+    reconstructions, else the above equations lose the regularization terms
+    resulting in inverse transform as solution.
+    The reconstruction in this case proceeds with a warning.
+    """
+
+    def __init__(self, fourier_op, linear_op=None, prox_op=None, mu=0,
+                 gradient_method="synthesis", lips_calc_max_iter=10,
+                 num_check_lips=10, lipschitz_cst=None, kspace_portion=0.1,
+                 smaps_extraction_mode='gridding',
+                 smaps_gridding_method='linear', n_jobs=1, verbose=0):
+        if linear_op is None:
+            linear_op = WaveletN(
+                wavelet_name="sym8",
+                nb_scale=3,
```

Comment:
Setting the `nb_scale` to 3  as a default value can be tricky due to the fact that this parameter depends on the reconstructed image shape

Reply:
I have added a TODO and this will be done as a part of #48 
We will obtain the value from shape seen in fourier op